### PR TITLE
feat(performance): update Souin setup & enable Cache-Control public in docs

### PR DESCRIPTION
**Pull request by laurent-bientz**
Hello,

This pull request updates Souin setup in docs, see https://github.com/dunglas/frankenphp/issues/976#issuecomment-2296050761

### CODE REVIEW

File: core/performance.md [hunk 1]
```diff
@@ -67,6 +75,8 @@ Setup the HTTP cache invalidation in your API Platform project
 ```yaml
 api_platform:
     http_cache:
+        # To make all responses public by default.
+        public: true
```
Comment (dunglas) [inline core/performance.md:79]: This a dangerous default, this should be done on a per-operation basis instead, except for entirely public APIs. Could you add an example on a per-resource basis instead (like this: https://api-platform.com/docs/core/performance/#setting-custom-http-cache-headers)?

We should also remove this default from the "Varnish" snippet.

**Review by dunglas — CHANGES_REQUESTED** [review]
Thank you very much for improving this, this is much appreciated! I left a comment regarding `public`, then we'll be ready to merge.

**Comment by dunglas** [comment]
By the way, the config can now be simplified:

Before:


```caddyfile
{
    order cache before rewrite
    ...
    cache {
        api {
            souin
        }
    }
}
```

After:


```caddyfile
{
    cache
    # ...
}

# ...
```

File: core/performance.md (hunk 1 above)
Comment (laurent-bientz) [reply core/performance.md]: `public: true` removed (also in Varnish) + example added

**Comment by laurent-bientz** [comment]
@dunglas I followed your recommendations, tell me if everything is ok

**Comment by dunglas** [comment]
Thank you!!

**Pull request merged**